File: 00-Методология научных исследований/2023/get_Pi/02.js
const get_pi_step = (count) => {
    let x, y;
    let count_circle = 0
    for (let step=1; step<=count; step++) {
        x = Math.random(), y = Math.random()
        if (x**2 + y**2 <= 1) {
            count_circle++
        }
        if (step%100_000_000 == 0) {
            console.log(step, 4*count_circle/step)
        }
    }
}

console.clear()
get_pi_step(1_000_000_000+1)


/*
____1_000_000 3.14_0328
___40_000_000 3.141_0718
1_000_000_000 3.14159_5552
*/

File: 00-Методология научных исследований/2023/get_Pi/02.py
from random import random


def get_pi(n=2):
    eps = 1 / 10**n
    count, count_circle = 0, 0
    diff = float('inf')
    last = 0
    while diff > eps:
        x, y = random(), random()
        if x**2 + y**2 <= 1:
            count_circle += 1
        count += 1
        pi = 4*count_circle/count
        diff = abs(last - pi)
        last = pi
        print(last)
        if count > 20: break
    return (count, pi)


n = int(input('Введите кол-во знаков после запятой - '))
print(f"Pi = {get_pi(n)}")
